src/test/java/redis/clients/jedis/tests/JedisPoolTest.java

```diff
@@ -381,4 +382,40 @@ private int getClientCount(final String clientList) {
     return clientList.split("\n").length;
   }
 
+  @Test
+  public void testResetInvalidPassword() {
+    JedisFactory factory = new JedisFactory(hnp.getHost(), hnp.getPort(), 2000, 2000, "foobared", 0, "my_shiny_client_name");
+    
+
+    try(JedisPool pool = new JedisPool(new JedisPoolConfig(), factory);
+        Jedis obj1 = pool.getResource();){
+      obj1.set("foo", "bar");
+      assertEquals("bar", obj1.get("foo"));
+      assertEquals(1, pool.getNumActive());
+      
+      factory.setPassword("wrong password");
+      try (Jedis obj2 = pool.getResource()) {
+        fail("Should not get resource from pool");
+      } catch (JedisConnectionException e) {}
+    }
+  }
+  
+  @Test
+  public void testResetValidPassword() {
+    JedisFactory factory = new JedisFactory(hnp.getHost(), hnp.getPort(), 2000, 2000, "bad password", 0, "my_shiny_client_name");
+    JedisPool pool = new JedisPool(new JedisPoolConfig(), factory);
+    Jedis obj = null;
+    try {
+      pool.getResource();
+      fail("Could not get resource from pool");
+    } catch (JedisConnectionException e) {
+      factory.setPassword("foobared");
+      obj = pool.getResource();
```

Comment:
Close `obj` within this catch block. Optionally, you can _declare_ `obj` (aka `Jedis obj = pool.getResource();`) here with/without try-with-resources.